### PR TITLE
Remove redundunt code, allow multiple DFA queries in one command

### PR DESCRIPTION
**Pull request by zhengyangl**

### CODE REVIEW

File: tools/souper-check.cpp [hunk 1]
```diff
@@ -134,87 +137,96 @@ int SolveInst(const MemoryBufferRef &MB, Solver *S) {
...
-    } else if (InferPowerTwo) {
-      bool PowTwo;
-      if (std::error_code EC = S->powerTwo(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
-                                           PowTwo, IC)) {
-        llvm::errs() << "Error: " << EC.message() << '\n';
-        Ret = 1;
-        ++Error;
-      } else {
+      if (InferPowerTwo) {
+        bool PowTwo;
```
Comment (regehr) [inline tools/souper-check.cpp]: do you see the darker green here? I think github is telling us about a whitespace problem, could you please check this?

Comment (zhengyangl) [reply tools/souper-check.cpp]: I don't see a problem here, it may be a diff glitch.

**Comment by zhengyangl** [comment]
Attached a test cases which shows different DFA queries can be called with a single command.

**Comment by jubitaneja** [comment]
LGTM, other than the whitespace issue.

File: tools/souper-check.cpp (hunk 1 above)
Comment (zhengyangl) [reply tools/souper-check.cpp]: It's actually my fault. Got it fixed.

File: tools/souper-check.cpp [hunk 2]
```diff
@@ -134,87 +137,96 @@ int SolveInst(const MemoryBufferRef &MB, Solver *S) {
...
-    } else if (InferRange) {
-      unsigned W = Rep.Mapping.LHS->Width;
-      llvm::ConstantRange Range = S->constantRange(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS, IC);
-
-      llvm::outs() << "known range from souper: " << "[" << Range.getLower()
-                   << "," << Range.getUpper() << ")" << "\n";
-      ++Success;
-    } else if (InferRHS || ReInferRHS) {
+    }
+    else if (InferRHS || ReInferRHS) {
```
Comment (regehr) [inline tools/souper-check.cpp]: please put else on the same line as its closing curly brace

Comment (regehr) [reply tools/souper-check.cpp]: also, if the "if" has curly braces, then the "else" must also

Comment (zhengyangl) [reply tools/souper-check.cpp]: fixed:)

File: tools/souper-check.cpp [hunk 3]
```diff
@@ -134,86 +137,94 @@ int SolveInst(const MemoryBufferRef &MB, Solver *S) {
...
+        } else {
+          llvm::outs() << "known signBits from souper: "
+                       << std::to_string(SignBits) << "\n";
+          ++Success;
+        }
       }
-    } else if (InferRange) {
-      unsigned W = Rep.Mapping.LHS->Width;
-      llvm::ConstantRange Range = S->constantRange(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS, IC);
+      if (InferRange) {
```
Comment (zhengyangl) [inline tools/souper-check.cpp]: these spaces are likely caused by diff glitch.

File: tools/souper-check.cpp [hunk 4]
```diff
@@ -134,86 +137,94 @@ int SolveInst(const MemoryBufferRef &MB, Solver *S) {
...
+    if (isInferDFA()) {
+      if (InferNeg) {
+        bool Negative;
+        if (std::error_code EC = S->negative(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
+                                             Negative, IC)) {
+          llvm::errs() << "Error: " << EC.message() << '\n';
+          Ret = 1;
+          ++Error;
+        } else {
+          llvm::outs() << "known negative from souper: "
```
Comment (regehr) [inline tools/souper-check.cpp]: please remove "known" from the description of any analysis results besides known bits

Comment (zhengyangl) [reply tools/souper-check.cpp]: fixed:)

Comment (zhengyangl) [reply tools/souper-check.cpp]: Oops, need to update test cases.

File: tools/souper-check.cpp (hunk 2 above)
Comment (regehr) [reply tools/souper-check.cpp]: wait, it doesn't look fixed
I am seeing this:
Screen Shot 2019-09-15 at 7 40 41 PM
and saying that it should look more like this:
```
} else if ...
```

**Comment by zhengyangl** [comment]
@regehr sorry John, seems that it is caused by a bad force push.

**Comment by regehr** [comment]
LGTM,thanks!

**Pull request merged**